jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/ViolationAssociationManager.java

```diff
@@ -35,96 +31,122 @@
 
 import java.io.BufferedReader;
 import java.io.InputStreamReader;
+import java.text.ParseException;
 import java.text.SimpleDateFormat;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 import java.util.stream.Collectors;
 
 /**
- * The Class ChildTableDataCollector.
+ * The Class ViolationAssociationManager.
  */
-public class ViolationAssociationManager implements Constants {
-
-    /** The Constant LOGGER. */
+public class ViolationAssociationManager {
     private static final Logger LOGGER = LoggerFactory.getLogger(ViolationAssociationManager.class);
-    
-	/** The s 3 account. */
-	private String s3Account = System.getProperty("base.account");
-	
-	/** The s 3 region. */
-	private String s3Region = System.getProperty("base.region");
-	
-	/** The s 3 role. */
-	private String s3Role =  System.getProperty("s3.role");
-	
-	/** The bucket name. */
-	private String bucketName =  System.getProperty("s3");
-	
-	/** The data path. */
-	private String dataPath =  System.getProperty("s3.data");
+
+    private static final String S3_ACCOUNT = System.getProperty("base.account");
+    private static final String S3_REGION = System.getProperty("base.region");
+    private static final String S3_ROLE = System.getProperty("s3.role");
+    private static final String BUCKET_NAME = System.getProperty("s3");
+    private static final String DATA_PATH = System.getProperty("s3.data");
+
+    private static final String DATE_FORMAT_NANO_SEC = "yyyy-MM-dd'T'HH:mm:ss.SSSSSSSSS'Z'";
+    private static final String DATE_FORMAT_MILL_SEC = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+    private static final String DATE_FORMAT_SEC = "yyyy-MM-dd HH:mm:00Z";
```

Comment:
@arunpaladin I remember we chatted about it and decided that UI needs to be changed so it can consume any amount of decimal digits... Or I'm missing something?